iceoryx_hoofs/test/moduletests/test_cxx_string.cpp

```diff
@@ -1547,8 +1571,10 @@ TYPED_TEST(stringTyped_test, CompareOperatorsWithEqualStringWithDifferentCapa)
     // required to verify string literal functionality of cxx::string
     // NOLINTNEXTLINE(hicpp-avoid-c-arrays, cppcoreguidelines-avoid-c-arrays)
     char testCharArray[STRINGCAP + 6U];
-    for (uint64_t i = 0U; i < STRINGCAP; ++i)
+    for (uint64_t i = 0U; i < STRINGCAP + 5U; ++i)
```

Comment:
IIRC the intention of this test was to compare a `cxx::string` with a larger char array which has the same content as the `cxx::string`. What was clang-tidy complaining about?

Reply:
That the array was not fully initialized when used.

Reply:
But there is an off by one error and the array is still not fully initialized. The last element is neither `\0` nor `m`. Therefore my suggestion to replace it with a ranged based for-loop

Reply:
Already done!